wrappers/python/nntile/layer/batch_norm.py

```diff
@@ -0,0 +1,163 @@
+# @copyright (c) 2022-present Skolkovo Institute of Science and Technology
+#                              (Skoltech), Russia. All rights reserved.
+#                2023-present Artificial Intelligence Research Institute
+#                              (AIRI), Russia. All rights reserved.
+#
+# NNTile is software framework for fast training of big neural networks on
+# distributed-memory heterogeneous systems based on StarPU runtime system.
+#
+# @file wrappers/python/nntile/layer/layer_norm.py
+# LayerNorm of NNTile Python package
+#
+# @version 1.0.0
+
+from operator import mul
+from functools import reduce
+
+from nntile.tensor import (
+    Tensor, TensorMoments,
+    prod_async, prod_slice_async, pow_async, sum_slice_async, sum_fiber_async, norm_slice_async, 
+    hypot_scalar_inverse_async, fill_async, add_slice_async, add_slice3_async,
+    add_fiber_async, add_scalar_async,
+    hypot_scalar_inverse_async,
+    prod_fiber_async, prod_fiber3_async,
+    add_async,
+
+    zeros, ones
+)
+
+from nntile.layer.base_layer import BaseLayer
+
+class BatchNorm2d(BaseLayer):
+    def __init__(self, x: TensorMoments, weight: Tensor, bias: Tensor, x_res_tm: TensorMoments, x_mean: Tensor, inv_std: Tensor, eps: float = 1e-05, redux: bool = False):
+        self.x_tm = x
+        self.x = x.value
+        self.eps = eps**0.5 # we will square it in forward (implementation constraints)
+
+        # TODO: add usage of learnable transforms 
+        self.weight = weight
+        self.bias = bias
+
+        self.x_res_tm = x_res_tm
+        self.x_res = self.x_res_tm.value
+
+        self.x_normalized_copy = zeros(self.x_res.shape, dtype=type(x.value))
```

Comment:
What is `type(value)`? I guess, `x.value` is either a tensor or `None`. Then you use a type or `None` as a dtype.

Reply:
Not a good idea to pass None to batchnorm. It will crash, it seems to be expected. 
We can add assert over not None - what do you think?

Reply:
I belive that the best option is to stick to `np.dtype` since we do not have as descriptive and complete type as `np.dtype`. Couple routines like `get_dtype(arr: Tensor)` and `get_tensor_ty(dtype: np.dtype`) together with `np.dtype` solves all interfacing issues until nntile has its own `dtype`.

Reply:
Your idea is clear and reasonable, at the same time It's very strange to require a user to use np.dtype to create nntile tensor. Second, we will have a lot numpy imports inside just for type usage

I tried to allow numpy and nntile types at the same type, it seems to be a reasonable compromise (see code)